src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/UpToDate/UpToDateCheckBuildEventNotifier.cs

```diff
@@ -113,7 +136,7 @@ private IEnumerable<IBuildUpToDateCheckProviderInternal> FindActiveConfiguredPro
                 }
             }
 
-            return Enumerable.Empty<IBuildUpToDateCheckProviderInternal>();
+            return null;
```

Comment:
This was changed to return `null` so that a consumer could avoid a JTF.RunAsync call when no providers exist.

Reply:
I don't like to use `Any` before enumerating. It's unclear whether you are double enumerating the source or not in such cases. For example, can you tell by looking at the code whether it is wasteful to enumerate the return value of `GetExportedValues` more than once?